internal/execution/execution.go

```diff
@@ -114,9 +116,22 @@ func call(ctx context.Context, url string, timeout time.Duration, body []byte) (
 	}
 	defer resp.Body.Close()
 
+	return handleResponse(resp)
+}
+
+func handleResponse(resp *http.Response) ([]byte, error) {
+	data, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
 	// Check for success between 200 and 299, redirect 300 to 399 is handled by the client, return error with statusCode >= 400
 	if resp.StatusCode >= 200 && resp.StatusCode <= 299 {
-		return io.ReadAll(resp.Body)
+		return data, nil
+	}
+
+	var parentErr error
+	if len(data) != 0 {
+		parentErr = errors.New(string(data))
 	}
-	return nil, zerrors.ThrowUnknown(nil, "EXEC-dra6yamk98", "Errors.Execution.Failed")
+	return nil, zhttp.HTTPStatusCodeToZitadelError(parentErr, resp.StatusCode, "EXEC-dra6yamk98", "Errors.Execution.Failed")
```

Comment:
i would prefer to have get the following responses:

```
zitadel response 400:
{
    parent: {
        status: 400,
        // one of the two bodies
        // body: 'Bad Request',
        // body: {"my": "value"},
    },
    message: "bli bla blub",
    id: "HTTP-2t49ulsjadf"
}

zitadel response 500:
{
    parent: {
        status: 99,
        // one of the two bodies
        // body: 'I got 99 problems but a server ain't one,
        // body: {"my": "value"},
    },
    message: "execution error",
    id: "HTTP-2t49ulsjadf"
}
```


The current implementation makes it too complicated for clients to parse the errors.

Reply:
IMO it's important to pass the body as is. If we need to introduce a new zerrors type for executions thats fine for me, if we cannot use an existing one.

Reply:
I used the existing ZitadelError basis type, with a separate ID, so that you can differentiate what the body tells. I also wrapped the body with '' to prevent missunderstandings.